metrics/anls/anls.py

```diff
@@ -0,0 +1,101 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""ANLS - Average Normalized Levenshtein Similarity"""
+
+import datasets
+from Levenshtein import ratio
+
+import evaluate
+
+_CITATION = """\
+@article{,
+    title = {Binary codes capable of correcting deletions, insertions, and reversals},
+    journal = {Soviet physics doklady},
+    volume = {10},
+    number = {8},
+    pages = {707--710},
+    year = {1966},
+    url = {https://nymity.ch/sybilhunting/pdf/Levenshtein1966a.pdf},
+    author = {V. I. Levenshtein},
+}
+"""
+
+_DESCRIPTION = """\
+ANLS refer to the average normalized Levenshtein similarity.
+"""
+
+_KWARGS_DESCRIPTION = """
+Computes Average Normalized Levenshtein Similarity (ANLS).
+Args:
+    predictions: List of question-answers dictionaries with the following key-values:
+    - 'question_id': id of the question-answer pair as given in the references
+    (see below)
+    - 'prediction_text': the text of the answer
+    references: List of question-answers dictionaries with the following key-values:
+    - 'question_id': id of the question-answer pair (see above)
+    - 'answers': list of possible texts for the answer, as a list of strings
+Returns:
+    'anls': The ANLS score of predicted tokens versus the gold answer
+Examples:
+    >>> predictions = [{'prediction_text': 'Denver Broncos',
+                        'question_id': '56e10a3be'}]
+    >>> references = [{'answers': ['Denver Broncos', 'Denver R. Broncos'],
+                       'question_id': '56e10a3be'}]
+    >>> anls_metric = evaluate.load("anls")
+    >>> results = anls_metric.compute(predictions=predictions, references=references)
+    >>> print(results)
+    {'anls_score': 100.0}
+"""
+
+
+@evaluate.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class Anls(evaluate.Metric):
+    def _info(self):
+        return evaluate.MetricInfo(
+            description=_DESCRIPTION,
+            citation=_CITATION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "predictions": {"question_id": datasets.Value("string"),
+                                    "prediction_text": datasets.Value("string")},
+                    "references": {
+                        "question_id": datasets.Value("string"),
+                        "answers": datasets.features.Sequence(datasets.Value("string")),
+                    },
+                }
+            )
+        )
+
+    def _compute_score(self, predictions, ground_truths):
+        theta = 0.5
+        anls_score = 0
+        for qid, prediction in predictions.items():
+            max_value = 0
+            if qid in ground_truths:
+                for x in ground_truths[qid]:
+                    nl = ratio(prediction, x)
```

Comment:
This implementation is wrong. `ratio` function from `python-Levenshtein==0.21.0` return 1 for equal strings!